File: examples/09_c_code/new.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    int nums[] = {1, 2, 3, 4, 5};
    int n = sizeof(nums) / sizeof(nums[0]);
    int sum = 0;

    for (int i = 0; i < n; i++) {
        sum += nums[i];
    }

    double avg = (double)sum / n;
    printf("Sum: %d\n", sum);
    printf("Average: %.2f\n", avg);

    return 0;
}

File: examples/08_rust_code/new.rs
use std::collections::HashMap;

fn main() {
    let nums = vec![1, 2, 3, 4, 5];

    // Calculate sum using fold
    let sum: i32 = nums.iter().fold(0, |acc, &x| acc + x);
    println!("Sum: {}", sum);

    // Calculate average
    let avg = sum as f64 / nums.len() as f64;
    println!("Average: {:.2}", avg);

    // Build a frequency map
    let mut freq: HashMap<i32, u32> = HashMap::new();
    for &n in &nums {
        *freq.entry(n).or_insert(0) += 1;
    }
    println!("Frequencies: {:?}", freq);
}
